src/cow-react/modules/limitOrders/containers/TradeButtons/limitOrdersTradeButtonsMap.tsx

```diff
@@ -35,16 +39,75 @@ export function SwapButton({
   )
 }
 
+const quoteErrorTexts: { [key in GpQuoteErrorCodes]: string } = {
+  [GpQuoteErrorCodes.UNHANDLED_ERROR]: 'Unhandled error',
+  [GpQuoteErrorCodes.TransferEthToContract]:
+    'Buying native currency with smart contract wallets is not currently supported',
+  [GpQuoteErrorCodes.UnsupportedToken]: 'Unsupported token',
+  [GpQuoteErrorCodes.InsufficientLiquidity]: 'Insufficient liquidity for this trade.',
+  [GpQuoteErrorCodes.FeeExceedsFrom]: 'Fees exceed from amount',
+  [GpQuoteErrorCodes.ZeroPrice]: 'Invalid price. Try increasing input/output amount.',
+}
+
 export const limitOrdersTradeButtonsMap: { [key in LimitOrdersFormState]: ButtonConfig | ButtonCallback } = {
-  [LimitOrdersFormState.LOADING]: {
+  [LimitOrdersFormState.Loading]: {
     disabled: true,
     text: 'Loading...',
   },
-  [LimitOrdersFormState.CAN_TRADE]: {
+  [LimitOrdersFormState.CanTrade]: {
     disabled: false,
     text: 'Review limit order',
   },
-  [LimitOrdersFormState.NOT_APPROVED]: ({ tradeState }: TradeButtonsParams) => {
+  // [LimitOrdersFormState.SwapIsUnsupported]: {
+  //   disabled: true,
+  //   text: 'Unsupported Token',
+  // },
+  [LimitOrdersFormState.WalletIsUnsupported]: {
+    disabled: true,
+    text: 'Wallet Unsupported',
+  },
+  [LimitOrdersFormState.ReadonlyGnosisSafeUser]: {
+    disabled: true,
+    text: 'Read Only',
+  },
+  [LimitOrdersFormState.NeedToSelectToken]: {
+    disabled: true,
+    text: 'Select a token',
+  },
+  [LimitOrdersFormState.AmountIsNotSet]: {
+    disabled: true,
+    text: 'Enter an amount',
+  },
+  [LimitOrdersFormState.InvalidRecipient]: {
+    disabled: true,
+    text: 'Enter a recipient',
+  },
+  [LimitOrdersFormState.CantLoadBalances]: {
+    disabled: true,
+    text: "Couldn't load balances",
+  },
+  [LimitOrdersFormState.QuoteError]: ({ quote }: TradeButtonsParams) => {
+    return (
+      <SwapButton disabled={true}>
+        <Trans>{quote.error ? quoteErrorTexts[quote.error.type] : 'Unknown error'}</Trans>
```

Comment:
I thin here you don't handle the case where you don't have the text for a given error. Right? 
For those cases, should we say: `Internal error: ${quote.error.type}`

Also, i would replace the `Unknown error` for a `Internal error`. Not strong opinion about the label though

Reply:
'Unknown error' - for the case when we can't detect the error, `quote.error` is undefined